### ISSUE TITLE
Only first subtitle is shown when 2nd subtitle timings start and end before the first subtitle ends.

### ISSUE THREAD

**Issue by Dnkhatri** [xbmc/xbmc#17609]
### Describe the bug
Here is a clear and concise description of what the problem is:

Only first subtitle is shown when 2nd subtitle timings start and end before the first subtitle ends.
example 
610
00:36:04,780 --> 00:36:12,980
[Bi Guan]

611
00:36:06,580 --> 00:36:09,860
He uses his real name? Boring.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

The 2nd line should show up below the first line as it does in vlc.

## Actual Behavior

Only the first [Bi Guan] shows up the 2nd line never shows up

### To Reproduce

1. 
2.
3.

### Debuglog

The debuglog can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Nvidia shield (Android 9)
 - Kodi version: 18.6 nightly

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by Dnkhatri** [comment]
I tried the same subtitles in mrmc and they work fine. Irc used to be the same for kodi I am not sure when it changed

**Comment by DaVukovic** [comment]
Coudl you please share more info?

Which kind of subtitle file?
Could you share that file so we are able to compare?
What about a full debuglog like we asked for at our template?

**Comment by Dnkhatri** [comment]
I have only tried it with srt subs. And the subtitle in question is this though. I have tried it with other srt subtitles with similar type of timings and they are also the same. It's easy to test just make a srt subtitle with first subtitle text 0:05-1:00 and 2nd text for 0:30-0:55 only the the first one shows up.

https://subscene.com/subtitles/my-lonely-planet-di-qiu-lian-hong-le-the-earth-blushes/english/2155736

There are portions where Chinese text is translated at the same time there is a conversation but only the subtitle that started earlier is shown. 
As far as debug goes I will try to post it later

**Comment by DaVukovic** [comment]
I downloaded that file and used it as a subtitle for a random movie (Ant-Man in that case) and I could partly reproduce your issue if I had Kodi in windowed mode. Being in windowed mode (and not in "Full Screen") I had to change the subtitle position to make those subtitles visible as you expect them to be. 

Anyway, I couldn't reproduce your issue while watching the movie in "Full Screen"-mode. 

First it looks like: 
![sub1](https://user-images.githubusercontent.com/7235787/78468366-abe3af00-7717-11ea-927b-d4aeb017952e.png)

Then the "[Bi Guan]" subtitle moves up and the other subtitle ("He uses his real name? Boring.") comes up: 
![sub2](https://user-images.githubusercontent.com/7235787/78468391-e1889800-7717-11ea-9b7b-f2c0b592fa96.png)

So, if that's different for you we will need: 

- screenshots
- a full debuglog

otherwise we can't help.

**Comment by Dnkhatri** [comment]
For me the subtitle moves up but 2nd subtitle does not show up. I reset the language settings to default to check if something wrong with my settings but that is not the case. On Android there is no separate windowed and full screen mode so can't test that. I will try different things to narrow the problem down then post the debug once I am sure as now I am curious why it is working for you and not for me.

**Comment by KarellenX** [comment]
I am using Windows 10 and a recent v18.7 nightly. I cannot reproduce this problem.

![sub1](https://user-images.githubusercontent.com/44626605/78469216-8f209900-7762-11ea-9940-f5ad6171b3bc.jpg)

![sub2](https://user-images.githubusercontent.com/44626605/78469221-95af1080-7762-11ea-810b-a2230553cb25.jpg)

**Comment by DaVukovic** [comment]
> I will try different things to narrow the problem down then post the debug once I am sure as now I am curious why it is working for you and not for me

Try to play with the subtitle position and see if that fixes something

**Comment by Dnkhatri** [comment]
> > I will try different things to narrow the problem down then post the debug once I am sure as now I am curious why it is working for you and not for me
> 
> Try to play with the subtitle position and see if that fixes something

Actually I think I found it in my first try this happens with 4k files. Tried it with a 1080p file it worked like it did with you used another 4k file it didnt.

**Comment by Dnkhatri** [comment]
https://paste.kodi.tv/ehonokagum
Here is the debug file first I played the 4k file then the 1080p file then another 4k file. I modified the first line in the sub to be till 40min so that it is easier to test.

**Comment by DaVukovic** [comment]
Yeah, I see it's a 4k movie. Have you tried to change the subtitle position. 

- Play the movie and pause it at the section the subtitle should be shown. (at 00:36:00.00)
- Open the VideoOSD, go to "settings" -> "Video settings" -> "Video calibration". 
- You will see something like that then:
![image](https://user-images.githubusercontent.com/7235787/78471450-ab581200-7731-11ea-9191-2c7d9b2580ef.png)
- Hit "Enter" until the bar at the bottom is highlighted:
![image](https://user-images.githubusercontent.com/7235787/78471471-d478a280-7731-11ea-992c-77805c23df69.png)
- then move the position "up" (I guess you can do that with your remote)

Does anything change then? Does the subtitle becomes visible?

**Comment by Dnkhatri** [comment]
I moved it up to middle of the screen. The first subtitle moves up and down at the correct time but the 2nd subtitle doesn't show.

**Comment by Dnkhatri** [comment]
I was able to reproduce the problem on my linux mint pc as well with the latest kodi 19 nightly. By changing the position of the subtitle to top of video I was able to find the 2nd subtitle it was at right side mostly being cut off. Only happens when playing 4k files in fullscreen. as the picture below shows hope you can reproduce and fix it. Its a weird bug that needs special set of circumstances to occur. 
![20200405_211515](https://user-images.githubusercontent.com/7855682/78499475-85f3f400-7783-11ea-8ba5-26f29ad16f1a.jpg)

**Comment by DaVukovic** [comment]
Are you able to test the same srt file on a 1080p movie? 

While reading that it seems that this makes a difference

**Comment by Dnkhatri** [comment]
> Are you able to test the same srt file on a 1080p movie?
> 
> While reading that it seems that this makes a difference

After trying out different things I am able to reproduce it when I set my pc desktop to 4k and use kodi in window mode with 4k and 1080p videos. In full screen as kodi resolution matches the video the subtitles work properly @1080p incorrectly @4k. When I set desktop mode to 1080p and then use windowed mode the subtitles work properly for both 1080p and 4k. When I disable resolution matching and force kodi to upscale 1080p to 4k and play 4k @ 4k the error occurs.

**Comment by Dnkhatri** [comment]
Was anyone able to reproduce this bug.

**Comment by Doraemoe** [comment]
I think I am having the same issue. I write a minimal subtitle file that could reproduce my problem:

```
[Script Info]
Title: [Test]
Original Script: Tester
ScriptType: v4.00+
WrapStyle: 0
PlayResX: 1280
PlayResY: 720
YCbCr Matrix: TV.709
ScaledBorderAndShadow: yes

[V4+ Styles]
Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, Encoding
Style: Default,Calibri,50,&H00FFFFFF,&HF0FFFFFF,&H007B66D0,&H808060E3,-1,0,0,0,100,100,0,0,1,2,1.5,2,10,10,13,1
Style: TOP,Candara,40,&H00FFFFFF,&HF0FFFFFF,&H00169CE1,&HF0FFFFFF,-1,0,0,0,100,100,0,0,1,1.5,0,9,10,12,8,128

[Events]
Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, Effect, Text
Dialogue: 0,0:00:01.00,0:00:09.00,TOP,,0,0,0,,{\be3\blur3\fad(200,200)}Should be on screen from 1s to 9s
Dialogue: 0,0:00:00.00,0:00:02.00,Default,,0,0,0,,{\be3}Should show/hide every 2s
Dialogue: 0,0:00:04.00,0:00:06.00,Default,,0,0,0,,{\be3}Should show/hide every 2s
Dialogue: 0,0:00:08.00,0:00:10.00,Default,,0,0,0,,{\be3}Should show/hide every 2s
```
Please save the above text into a `Test.ass` file and play with any video.

This subtitle has a line shown on screen at upper right between 1s to 9s. In addition, it has another subtitle shown at normal position every 2 seconds between 0 to 10s. When Kodi plays it, the subtitle on the upper right will disappear between 6s to 8s while it should be on screen all the time. This has been tested on Kodi 18.8 Windows and CoreELEC.

This subtitle plays normally with MPC on Windows.
